### PR TITLE
Join existing list token

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person.
Tokens can be shared with other users to allow them to co-manage a given list.

Joining an existing list consists of the following:

* User enters an existing token
* Save the token to `localStorage`
* Show the user the existing list

AC:

* When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list
Notes:
* As of this story, the welcome/home screen will be feature complete, as reflected in this wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
* For accessibility reasons, use a `<label>` tag for the "share token" field.
* You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
* If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue 5).

### PR DESCRIPTION
**Pull request by hellodeborahuk**
## Description

- This PR creates a form to enter an existing token. 
- Created `checkIfTokenExists` to query a document from the firestore database with the token that the user entered. 
- If if doesn't find the document an error is displayed, else it sets the token and saves it to local storage. 

### Context for global state token

- Created `TokenContext` file, and defined inside of it a custom provider called `TokenProvider` and wrap our App.js with it.
- `TokenProvider` provides `token` and `setToken` to the whole app.
- Created a custom hook `useToken` that will consume the context.
- Replaced token useState with the global state provided by `TokenContext`.

## Related Issue

Close #5 

## Acceptance Criteria

- When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
- Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
- On submit, show an error if the token does not exist
- If they get an error message, allow them to try again or create a new list

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
- From your terminal, pull down this branch with git pull origin `rb-dd-join-existing-list` and check that branch out with git checkout `rb-dd-join-existing-list`.
- Then npm to install the newly added dependencies locally and npm start to launch the app.
- Use "mac jail astral" token to join an existing list.
- Enter an incorrect token to see the error message.

### CODE REVIEW

**Comment by hellodeborahuk** [comment]
Updated the copy on the home page and added "share token" to the label on the form.

**Comment by redapy** [comment]
Thank you @hellodeborahuk for adding the missing label!

File: src/components/ShareTokenForm.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    } else {
+      setError(null);
+      localStorage.setItem('token', selectedToken);
+      setToken(selectedToken);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="form">
+      <label htmlFor="shareToken">
```
Comment (JimeBlue) [inline src/components/ShareTokenForm.js]: It´s great that you took accesability into account!

Comment (JimeBlue) [reply src/components/ShareTokenForm.js]: What a wonderfull job you did @redapy and @hellodeborahuk! Not only it works very well, but also you have produced a very clean, easy to follow and efficient code. Congratulations!!!

File: src/components/ShareTokenForm.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react';
+import { collection, getDocs, query, where } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import useToken from '../hooks/useToken';
+
+const ShareTokenForm = () => {
+  const [error, setError] = useState(null);
```
Comment (JimeBlue) [inline src/components/ShareTokenForm.js]: Myabe `error ` could have a more self explanatory name, something that shows the purpose of the error, something in the lines of incorrect token error or non exsiting token error. This is a tiny suggestion and a matter of personal preference.

File: src/hooks/useToken.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+const useToken = () => {
+  const value = useContext(TokenContext);
+  if (value === undefined) {
+    console.log('Error');
+  }
+  return value;
+};
+
+export default useToken;
```
Comment (Amaka202) [inline src/hooks/useToken.js:12]: I love how you used a custom hook here. Beautiful implementation!

**Review by Amaka202 — APPROVED** [review]
Good job team!🎉💯🔥 I loved your implementation a lot. 

A little non-blocking suggestion is that we may need to have a loading state for when the items are still fetching from the database and when the items are being sent to the database. Also, have an error state for when the fetching fails perhaps because of users' network issues or any order cause.

It will improve the user experience a lot!👍👍

Finally, it would be nice to wipe the error message after a few seconds too.

**Pull request merged**